components/kyma-environment-broker/cmd/cli/skr/options.go

```diff
@@ -0,0 +1,227 @@
+	kebAPIURL          string
+	kubeconfigAPIURL   string
+	gardenerKubeconfig string
+}
+
+// GlobalOpts is the convenience object for storing the fixed global conifguration (parameter) keys
+var GlobalOpts = GlobalOptionsKey{
+	oidcIssuerURL:      "oidc-issuer-url",
+	oidcClientID:       "oidc-client-id",
+	oidcClientSecret:   "oidc-client-secret",
+	kebAPIURL:          "keb-api-url",
+	kubeconfigAPIURL:   "kubeconfig-api-url",
+	gardenerKubeconfig: "gardener-kubeconfig",
+}
+
+// SetGlobalOpts configures the global parameters on the given root command
+func SetGlobalOpts(cmd *cobra.Command) {
+	cmd.PersistentFlags().String(GlobalOpts.oidcIssuerURL, "", "OIDC authentication server URL to use for login. Can also be set the SKR_OIDC_ISSUER_URL environment variable")
+	viper.BindPFlag(GlobalOpts.oidcIssuerURL, cmd.PersistentFlags().Lookup(GlobalOpts.oidcIssuerURL))
+
+	cmd.PersistentFlags().String(GlobalOpts.oidcClientID, "", "OIDC client ID to use for login. Can also be set via the SKR_OIDC_CLIENT_ID environment variable")
+	viper.BindPFlag(GlobalOpts.oidcClientID, cmd.PersistentFlags().Lookup(GlobalOpts.oidcClientID))
+
+	cmd.PersistentFlags().String(GlobalOpts.oidcClientSecret, "", "OIDC client Secret to use for login. Can also be set via the SKR_OIDC_CLIENT_SECRET environment variable")
+	viper.BindPFlag(GlobalOpts.oidcClientSecret, cmd.PersistentFlags().Lookup(GlobalOpts.oidcClientSecret))
+
+	cmd.PersistentFlags().String(GlobalOpts.kebAPIURL, "", "Kyma Environment Broker API URL to use for all commands. Can also be set via the SKR_KEB_API_URL environment variable")
+	viper.BindPFlag(GlobalOpts.kebAPIURL, cmd.PersistentFlags().Lookup(GlobalOpts.kebAPIURL))
+
+	cmd.PersistentFlags().String(GlobalOpts.kubeconfigAPIURL, "", "OIDC Kubeconfig Service API URL, used by the skr kubeconfig and taskrun commands. Can also be set via the SKR_KUBECONFIG_API_URL environment variable")
+	viper.BindPFlag(GlobalOpts.kubeconfigAPIURL, cmd.PersistentFlags().Lookup(GlobalOpts.kubeconfigAPIURL))
+
+	cmd.PersistentFlags().String(GlobalOpts.gardenerKubeconfig, "", "Path to the corresponding Gardener project kubeconfig file which have permissions to list/get shoots. Can also be set via the SKR_GARDENER_KUBECONFIG environment variable")
+	viper.BindPFlag(GlobalOpts.gardenerKubeconfig, cmd.PersistentFlags().Lookup(GlobalOpts.gardenerKubeconfig))
+}
+
+// ValidateGlobalOpts checks the presence of the required global configuration parameters
+func ValidateGlobalOpts() error {
+	var reqGlobalOpts = []string{GlobalOpts.oidcIssuerURL, GlobalOpts.oidcClientID, GlobalOpts.oidcClientSecret, GlobalOpts.kebAPIURL}
+	var missingGlobalOpts []string
+	for _, opt := range reqGlobalOpts {
+		if viper.GetString(opt) == "" {
+			missingGlobalOpts = append(missingGlobalOpts, opt)
+		}
+	}
+
+	if len(missingGlobalOpts) == 0 {
+		return nil
+	}
+	return fmt.Errorf("missing required options: %s. See skr --help for more information", strings.Join(missingGlobalOpts, ", "))
```

Comment:
IMO we need some documentation with examples of using that CLI, including login to the cluster

Reply:
I was thinking of that too, this is my idea:

1. For general usage, generate Markdown docs under docs/skr-cli similar to [kyma cli gendocs](https://github.com/kyma-project/cli/tree/master/docs) or using the builtin [cobra Makrdown generator](https://github.com/spf13/cobra/blob/master/doc/md_docs.md) 
2. For our Kyma Control Plane specific use, we will create internal doc with config files having the required params (OIDC client ID/secret, API URLs)

Reply:
I like that idea, in the open-source, we can also include some overview readme about basic usage of the CLI because, for example, I don't know how to display examples that are included in the command definitions, if that's even possible.